lib/src/connector/course_connector.dart

```diff
@@ -101,38 +101,41 @@ class CourseConnector {
       //取得學期資料
       nodes = courseNodes[0].getElementsByTagName("td");
       SemesterJson semester = SemesterJson();
-      semester.year = nodes[1].text;
-      semester.semester = nodes[2].text;
+
+      final titleString = nodes[0].text;
+      RegExp regex = RegExp(r'\b\d+\b');
+      Iterable<RegExpMatch> matches = regex.allMatches(titleString);
+      List<String> numbers = matches.map((match) => match.group(0)).toList();
+      semester.year = numbers[1];
+      semester.semester = numbers[2];
 
       courseExtraInfo.courseSemester = semester;
 
       CourseExtraJson courseExtra = CourseExtraJson();
 
-      courseExtra.name = nodes[3].getElementsByTagName("a")[0].text;
-      if (nodes[3].getElementsByTagName("a")[0].attributes.containsKey("href")) {
-        courseExtra.href = _courseCNHost + nodes[3].getElementsByTagName("a")[0].attributes["href"];
-      }
-      courseExtra.category = nodes[7].text; // 取得類別
-      courseExtra.openClass = nodes[9].text;
-      courseExtra.selectNumber = nodes[11].text;
-      courseExtra.withdrawNumber = nodes[12].text;
-      courseExtra.id = courseId;
+      nodes = courseNodes[1].getElementsByTagName("tr");
+      List<String> courseIds = nodes.skip(2).map((node) => node.getElementsByTagName("td")[0].text).toList();
+      node = nodes[courseIds.indexWhere((element) => element.contains(courseId)) + 2];
```

Comment:
```suggestion
      final courseIdPosition = courseIds.indexWhere((element) => element.contains(courseId));
      if (courseIdPosition == -1) {
        // do something...
      } else {
        node = nodes[courseIdPosition + 2];
      }
```

Suggest always avoiding doing complex operation in a `[]` operator.
In this case, the `indexWhere` method would return `-1` instead of throwing an exception. This may cause `-1 + 2` to `1`, which makes un-expected indexing and silent error showing.

Reply:
[25353ae](https://github.com/NEO-TAT/tat_flutter/pull/214/commits/25353ae021222e3371e921e9407edaafde7e1f94)

Reply:
How about the case when `courseIdPosition` is `-1` ? should the rest of process be continued ?

Reply:
similar to the reason:
> should i issue an error directly when the numbers filter isn't working properly? i think the try-catch block is dealing with exception of these parsers already

now it will throw StateError for CourseId not found (courseIdPosition is read as -1)
 [16339a9](https://github.com/NEO-TAT/tat_flutter/pull/214/commits/16339a95622257c57026b546bf55dac52b0925c1)